packages/webapp/src/containers/Animals/Inventory/useAnimalInventory.tsx

```diff
@@ -0,0 +1,343 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import { FC, useMemo } from 'react';
+import i18n from '../../../locales/i18n';
+import {
+  useGetAnimalsQuery,
+  useGetAnimalBatchesQuery,
+  useGetAnimalGroupsQuery,
+  useGetCustomAnimalBreedsQuery,
+  useGetCustomAnimalTypesQuery,
+  useGetDefaultAnimalBreedsQuery,
+  useGetDefaultAnimalTypesQuery,
+} from '../../../store/api/apiSlice';
+import {
+  Animal,
+  AnimalBatch,
+  AnimalGroup,
+  CustomAnimalBreed,
+  CustomAnimalType,
+  DefaultAnimalBreed,
+  DefaultAnimalType,
+} from '../../../store/api/types';
+import { getComparator, orderEnum } from '../../../util/sort';
+import { ReactComponent as CattleIcon } from '../../../assets/images/animals/table/cattle.svg';
+import { ReactComponent as ChickenIcon } from '../../../assets/images/animals/table/chicken.svg';
+import { ReactComponent as PigIcon } from '../../../assets/images/animals/table/pig.svg';
+import { ReactComponent as BatchIcon } from '../../../assets/images/animals/table/batch.svg';
+import { AnimalTranslationKey } from '../types';
+
+export type AnimalInventory = {
+  icon: FC;
+  identification: string;
+  type: string;
+  breed: string;
+  groups: string[];
+  path: string;
+  count: number;
+};
+
+const { t } = i18n;
+
+const getDefaultAnimalIcon = (defaultAnimalTypes: DefaultAnimalType[], defaultTypeId: number) => {
+  const key = defaultAnimalTypes[defaultTypeId].key;
+  switch (key) {
+    case AnimalTranslationKey.CATTLE:
+      return CattleIcon;
+    case AnimalTranslationKey.CHICKEN_BROILERS:
+      return ChickenIcon;
+    case AnimalTranslationKey.CHICKEN_LAYERS:
+      return ChickenIcon;
+    case AnimalTranslationKey.PIGS:
+      return PigIcon;
+    default:
+      return CattleIcon;
+  }
+};
```

Comment:
This can be improved using an object.
```
const defaultAnimalIcon = {
  [AnimalTranslationKey.CATTLE]: CattleIcon,
  [AnimalTranslationKey.CHICKEN_BROILERS]: ChickenIcon,
  ...
}
```

Reply:
I agree in principle. Lookup tables being slightly more efficient. But for similar (but different) components choice situations it was helful for integration with typescript to go this route. https://github.com/LiteFarmOrg/LiteFarm/pull/3138#discussion_r1498735620

I acknowledge no props are being passed and the so the situation is different but I thought there was some benefit in keeping the structure the same for future copy-pasters like myself.